### DISCUSSION TITLE
Restart browser only for a specific test file?

### DISCUSSION THREAD

**Discussion by socketopp** [codeceptjs/CodeceptJS#3638] [Q&A] [answered]
Is this possible?

```
exports.config = {
  helpers: {
    Playwright: {
      restart: false,
       }
   }
...
  mocha: {
      grep: '@restart',
       only: true,
  }

};
```

Been tagging my Feature and Scenarios in that test file with @restart and it continues not to restart the browser when it begins on a new test file. 

Any other ideas on how to achieve this?

**Comment by adudyak** [comment]
Browser is restarted between tests(between your _test.js/ts files). I assume,` restart: "true"` is default setting since CodeceptJS 3.1
You can additionally restart tests within same test file with method `I.restartBrowser()`. Put it in Before/After suites
Mocha is reporter, it won't restart browser.

**Reply by socketopp** [reply]
Sorry if I was unclear, my aim is to restart all tests between scenarios except scenarios in one specific test file. Not sure how I can accomplish that.

**Reply by adudyak** [reply, marked as answer]
`first_test.js` Browser will be restarted before each scenario:
```
Feature("restarted")

Before(({ I }) =>{
I.restartBrowser(contextOptions);
});

Scenario("1");
Scenario("2");
```

`second_test.js` Browser won't be restarted while running these scenarios:
```
Feature("not restarted");

Scenario("1");
Scenario("2");
```

If you want to not restart browser between `first_test.js` and `second_test.js`, then set `restart: false` in settings.
https://codecept.io/helpers/Playwright/#restartbrowser
